src/test/java/com/exadel/discountwebapp/integrationtest/service/RoleServiceImplIntegrationTest.java

```diff
@@ -0,0 +1,79 @@
+package com.exadel.discountwebapp.integrationtest.service;
+
+import com.exadel.discountwebapp.DiscountWebApplication;
+import com.exadel.discountwebapp.role.entity.Role;
+import com.exadel.discountwebapp.role.mapper.RoleMapper;
+import com.exadel.discountwebapp.role.repository.RoleRepository;
+import com.exadel.discountwebapp.role.service.RoleService;
+import com.exadel.discountwebapp.role.vo.RoleResponseVO;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.jdbc.EmbeddedDatabaseConnection;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.annotation.Rollback;
+
+import javax.transaction.Transactional;
+import java.time.LocalDateTime;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@Transactional
```

Comment:
We can clean up database after every test with SQL cleanup script (@Sql( ... cleanup.sql ))